### ISSUE TITLE
Computation of  Langevin dynamics

### ISSUE THREAD

**Issue by yxsysu** [ChenWu98/unified-generative-zoo#1]
In the [line100](https://github.com/ChenWu98/unified-generative-zoo/blob/main/model/langevin_dynamics.py#L100) in langevin_dynamics.py, the z is updated according to the Langevin dynamics:
`z = (z - self.step_size / 2 * grad + noise).detach()`.
The variable `grad' in code is computed as $\bigtriangledown_z -E(G(z)|C)$.

Refer to Sec 3.4 【UNIFIED PLUG-AND-PLAY GUIDANCE FOR GENERATIVE MODELS】 in paper 【UNIFYING DIFFUSION MODELS’ LATENT SPACE, WITH APPLICATIONS TO CYCLEDIFFUSION AND GUIDANCE】, the update of z should involve the term $\bigtriangledown_z  \log(p_z(z^k))$. 
 
I am wondering why this term can be omitted in practice or which codes compute this term. Can you give me some hints?  Thank you for your great work and your kindly reply.

**Comment by ChenWu98** [comment]
In the code implementation, I merged p(z) into the energy function (cf. [PriorZEnergy](https://github.com/ChenWu98/unified-generative-zoo/blob/5d2b8b91fc641be3b03216405f79f0cc97ac7d89/config/experiments/clip_an_old_person_ffhq512_styleganxl_langevin_700.cfg#L23)).

**Comment by yxsysu** [comment]
Thank you for your reply! I get it!
